enbeec/rgblent-client
# Palette Data

## CHANGES:
- `src/RGBlent.js`
	- tweaked layout styling
	- `<Palette />` now renders it's own `<Row />`
- `src/components/ColorProvider.js`
	- added `getPalette` and `CURRENT_PALETTE` query key (`KEYS`)
	- `setColor` now automatically triggers a refetch
- `src/components/Detail.js`
	- styling changes
	- when the `Accordion` section "RGB" is open, the hex code from `colorInfo` is displayed
	- the hex code for the state color to the swatch column
	- tweaked render logic
- `src/components/Picker.js`
	- styling tweaks
	- simplified load function thanks to new `setColor` in `ColorProvider`
- `src/components/Swatch.js`
	- now accepts extra dropdown menu items via props
- `src/css/HexColorPicker.css`
	- commented out scale -- I like the size but it's the wrong way to do it
- `src/utils/auth.js`
	- added `IsNobody` predicate (function that returns a `bool`)
- `src/utils/color.js`
	- added `DEFAULT_PALETTE_MINIMAL` data for query initialization
		- matches the "/default/palette" return enough to satisfy rendering
- `src/components/Palette`
	- big styling overhaul
	- state for the name of the current palette (default is "default")
	- state for each color cell (default is the minimal default palette constant)
	- implemented a query to get a palette
		- uses a compound query key that starts with a constant and ends with the palette name
		- never goes stale until manually refetched
		- onSuccess it sets the palette color state
	- palette colors are rendered as cards
		- if a user is logged in:
			- if the color slot matches it's corresponding palette slot it is "clean"
				- the slot's header will be the corresponding palette color's label
				- there is a Replace button that loads the state color into that palette slot
					- that slot is then marked as "dirty"
					- any other "dirty" slot is reverted to the palette color
			- if it is "dirty"
				- the slot's header will be the color's hex code
				- the Replace button becomes the Restore button and reverts that slot to the palette color
				- the hex code next to the Replace/Restore button becomes a Save button (currently no-op)
		- if nobody is logged in:
			- the Replace button is disabled and the card title will always be the palette color label
			- a tooltip informs the user as to why the button is disabled
	- the palette's name is displayed as a header
		- if anything is dirty, an asterisk appears next to the name

## TO TEST:
- hard to test a lot of the card slot functionality now that I added the Nobody check
- registration/login is next up so those features will be back soon

## RELATED ISSUES:

Closes #4
